File: frontend/src/types/Question.ts
export type Difficulty = 'easy' | 'medium' | 'hard'
export type AnswerOption = 'A' | 'B' | 'C' | 'D'

export interface QuestionPublic {
  id: number
  topic: string
  question_text: string
  option_a: string
  option_b: string
  option_c: string
  option_d: string
  difficulty: Difficulty
}

export interface QuestionWithAnswer extends QuestionPublic {
  correct_answer: AnswerOption
  explanation: string
}

export interface TopicSummary {
  topic: string
  question_count: number
  difficulties: Difficulty[]
}

File: backend/app/schemas/question.py
from pydantic import BaseModel, ConfigDict

from app.models.question import AnswerOption, Difficulty


class QuestionPublic(BaseModel):
    """Question shape shown to a student while a quiz is in progress (no answer/explanation)."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    topic: str
    question_text: str
    option_a: str
    option_b: str
    option_c: str
    option_d: str
    difficulty: Difficulty


class QuestionWithAnswer(QuestionPublic):
    correct_answer: AnswerOption
    explanation: str
    source_document_id: int | None = None


class TopicSummary(BaseModel):
    topic: str
    question_count: int
    difficulties: list[Difficulty]
